packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -0,0 +1,500 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic_types.dart';
+
+/// [StrutStyle] defines the strut, which defines the minimum height a line can be
+/// relative to the baseline. Strut applies to all lines in the pararaph.
+///
+/// Strut is a feature that allows minimum line heights to be set. The effect is as
+/// if a zero width space was included at the beginning of each line in the
+/// paragraph. This imaginary space is 'shaped' according the properties defined
+/// in this class.
+///
+/// No lines may be shorter than the strut. The ascent and descent of the strut
+/// are calculated, and any laid out text that has a shorter ascent or descent than
+/// the strut's ascent or descent will take the ascent and descent of the strut.
+/// Text with ascents or descents larger than the strut's ascent or descent will lay
+/// out as normal and extend past the strut.
+///
+/// Strut is defined independently from any text content or [TextStyle]s.
+///
+/// The vertical components of strut are as follows:
+/// 
+///  * `leading * fontSize / 2` or half the font leading if `leading` is undefined (half leading)
+///  * `ascent * height`
+///  * `descent * height`
+///  * `leading * fontSize / 2` or half the font leading if `leading` is undefined (half leading)
+///
+/// The sum of these four values is the total height of the line.
+///
+/// The `ascent + descent` is equivalent to the [fontSize]. Ascent is the font's
+/// spacing above the baseline without leading and descent is the spacing below the
+/// baseline without leading. Leading is split evenly between the top and bottom.
+/// The values for `ascent` and `descent` are provided by the font named by
+/// [fontFamily]. If no [fontFamily] or [fontFamilyFallback] is provided, then the
+/// platform's default family will be used.
+///
+/// Each line's spacing above the baseline will be at least as tall as the half
+/// leading plus ascent. Each line's spacing below the baseline will be at least as
+/// tall as the half leading plus descent.
+/// 
+/// ### Fields and their default values.
+
+// ///////////////////////////////////////////////////////////////////////////
+// The defaults are noted here for convenience. The actual place where they //
+// are defined is in the engine paragraph_style.h of LibTxt. The values here//
+// should be updated should it change in the engine. The engine specifies   //
+// the defaults in order to reduce the amount of data we pass to native as  //
+// strut will usually be unspecified.                                       //
+// ///////////////////////////////////////////////////////////////////////////
+
+///
+/// Omitted or null properties will take the default values specified below:
+///
+///  * [fontFamily] the name of the font to use when calcualting the strut (e.g., Roboto).
```

Comment:
This will render as follows:

```
  * fontFamily the name of the font
```

Put a colon or hyphen or something between the property name and the description.